worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockTypes.java

```diff
@@ -825,6 +827,24 @@ public BlockState withPropertyId(int propertyId) {
         return IntStream.of(settings.stateOrdinals).filter(i -> i != -1).mapToObj(i -> states[i]).collect(Collectors.toList());
     }
 
+    public BlockState getState(Map<Property<?>, Object> key) {
```

Comment:
This was the function I had to implement myself to be compliant with the API change to BlockType.java from upstream. 

Hopefully this will work, but I have no way to test it.